src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/MethodWithGCInfo.cs

```diff
@@ -42,6 +43,11 @@ public void SetCode(ObjectData data)
             _methodCode = data;
         }
 
+        public bool AlreadyStartedCompilation()
```

Comment:
We don't need this - the part that deals with possibly compiling the same method multiple times is unreachable in this codebase. The code that tries to handle compiling the same method twice was copy-pasted from the CoreRT full AOT compiler, but we don't use `ShadowConcreteMethod` in this compiler (which is the only reason why we could end up in that situation).

Feel free to delete the `ShadowConcreteMethodNode` cast in ReadyToRunCodegenCompilation.cs, this method, and the callsite. We don't need to worry about compiling the same method twice.

There's another cast to `ShadowConcreteMethodNode` in ReadyToRunTableManager.cs that is also unreachable (can be deleted too).